homeassistant/components/proxmoxve/config_flow.py

```diff
@@ -0,0 +1,231 @@
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .common import ResourceException
+from .const import (
+    CONF_CONTAINERS,
+    CONF_NODE,
+    CONF_NODES,
+    CONF_REALM,
+    CONF_VMS,
+    DEFAULT_PORT,
+    DEFAULT_REALM,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_REALM, default=DEFAULT_REALM): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_VERIFY_SSL, default=False): cv.boolean,
+    }
+)
+
+
+async def _validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input and fetch data."""
+
+    def create_proxmox_client() -> ProxmoxAPI:
+        """Create a ProxmoxAPI client."""
+        user_id = (
+            data[CONF_USERNAME]
+            if "@" in data[CONF_USERNAME]
+            else f"{data[CONF_USERNAME]}@{data[CONF_REALM]}"
+        )
+
+        return ProxmoxAPI(
+            data[CONF_HOST],
+            port=data[CONF_PORT],
+            user=user_id,
+            password=data[CONF_PASSWORD],
+            verify_ssl=data.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL),
+        )
+
+    try:
+        client = await hass.async_add_executor_job(create_proxmox_client)
+    except AuthenticationError as err:
+        raise ProxmoxAuthenticationError from err
+    except SSLError as err:
+        raise ProxmoxSSLError from err
+    except ConnectTimeout as err:
+        raise ProxmoxConnectTimeout from err
+
+    try:
+        nodes = await hass.async_add_executor_job(client.nodes.get)
+    except (ResourceException, requests.exceptions.ConnectionError) as err:
+        raise ProxmoxNoNodesFound from err
+
+    _LOGGER.debug("Proxmox nodes: %s", nodes)
+
+    nodes_data: list[dict[str, Any]] = []
+    for node in nodes:
+        try:
+            vms = await hass.async_add_executor_job(client.nodes(node["node"]).qemu.get)
+            containers = await hass.async_add_executor_job(
+                client.nodes(node["node"]).lxc.get
+            )
+        except (ResourceException, requests.exceptions.ConnectionError) as err:
+            raise ProxmoxNoNodesFound from err
+
+        nodes_data.append(
+            {
+                "node": node["node"],
+                "vms": vms,
+                "containers": containers,
+            }
+        )
+
+    _LOGGER.debug("Nodes with data: %s", nodes_data)
+    return {"nodes": nodes_data}
```

Comment:
```suggestion
    return nodes_data
```

We don't need to return a dict if only one element. Instead we can return the data directly